core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -710,9 +711,51 @@ class TiBatchWrite(@transient val df: DataFrame,
     tiTableInfo.getId
   }
 
+  private def estimateRegionSplitNum(wrappedRowRdd: RDD[WrappedRow]) = {
+    //TODO refine this https://github.com/pingcap/tispark/issues/891
+    val rowSize = tiTableInfo.getEstimatedRowSizeInByte
+    //TODO: replace 96 with actual value read from pd https://github.com/pingcap/tispark/issues/890
+    (wrappedRowRdd.count() * rowSize) / (96 * 1024 * 1024)
+  }
+
+  private def splitIndexRegion(wrappedRowRdd: RDD[WrappedRow]) = {
+    if (options.enableRegionSplit && isEnableSplitRegion) {
+      val regionSplitNum = if (options.regionSplitNum != 0) {
+        options.regionSplitNum
+      } else {
+        estimateRegionSplitNum(wrappedRowRdd)
+      }
+
+      val sampledDataList =
+        wrappedRowRdd.takeSample(withReplacement = false, num = regionSplitNum.toInt).toList
```

Comment:
is it possible that 
1. calculate `min` and `max`
2. evenly divided
just like region split for table?

Reply:
It is not easy to do that right now. We have to know the min max of each column which we have not implemented yet.

Reply:
It is not feasible to do that at this point, we have to first implement the min and max of each column type.

Reply:
can we use spark's `min` and `max` ？

Reply:
Each column type has it is own min and max. For example, if we have a time column, the min of this column is ` -838:59:59` and the max is `838:59:59`. Using spark's min and max is not relevant here.

Reply:
user interface is DataFrame, which means all user provided data is spark sql supported type, which means we can use spark `min` and `max`.

Reply:
That requires implementing orders which is something we do not have right now.

Reply:
can we calculate `min` and `max` after encoding?